### DISCUSSION TITLE
local,"could not establish connection

### DISCUSSION THREAD

**Discussion by lecavalie** [zubkov-andrei/pg_profile#70] [Q&A] [answered]
На одном сервере все работает, на втором попробовал, вылезает такая ошибка. Будьте любезны, подскажите в каком направлении искать решение этой ошибки.

postgres=# select take_sample();
                                                    take_sample                                                    
-------------------------------------------------------------------------------------------------------------------
 (local,"could not establish connection                                                                           +
 SQL-оператор: ""SELECT dblink_connect('server_connection', server_properties #>> '{properties,server_connstr}')""+
 функция PL/pgSQL init_sample(integer), строка 60, оператор PERFORM                                               +
 функция PL/pgSQL take_sample(integer,boolean), строка 14, оператор присваивание                                  +
 функция PL/pgSQL take_sample_subset(integer,integer), строка 27, оператор присваивание                           +
 SQL-функция ""take_sample"", оператор 1                                                                          +
 ВАЖНО:  пользователь ""profile_mon"" не прошёл проверку подлинности (Peer)",00:00:00)
(1 строка)

**Comment by zubkov-andrei** [comment]
Добрый день!
Тут всё связано исключительно с тем, что профайл не может зайти в базу данных по dblink'у. Ключевое сообщение об ошибке "пользователь ""profile_mon"" не прошёл проверку подлинности (Peer)"
Оно говорит о том, что для локальный соединений у вас настроена peer авторизация в pg_hba.conf. Это правильно. Но подключение вы пытаетесь выполнить от имени пользователя "profile_mon". Поскольку подключение выполняется бэкендом Postgres, пользователь ОС, от которого оно произволится - postgres. В результате, авторизация не проходит.

**Reply by lecavalie** [reply]
Как лучше решить это? Настройки дефолтные, пока не понимаю, почему на 1 сервере у меня все прошло без этой ошибки

**Reply by zubkov-andrei** [reply]
Всё зависит от условий, в которых вам необходимо сипользовать profile. Самый простой вариант - выполнять снимки от пользователя postgres, тогда у вас peer авторизация будет проходить успешно.
Посмотрите строку подключения, используемую для локального сервера в обоих ваших экземплярах. profile покажет ее через функцию show_servers().
По идее, используя указанную строку подключения у вас должно получиться выполнить подключение через dblink_connect. Если не получится у вас, не получится и у profile'а.

**Reply by lecavalie** [reply]
postgres=# select show_servers();
                               show_servers                                
---------------------------------------------------------------------------
 (local,"dbname=postgres port=5432 user=profile_mon password=pwd_mon",t,,)
(1 строка)

postgres=#

**Reply by zubkov-andrei** [reply]
Если у вас на обоих серверах одинаковая строка подключения, сравните файлы pg_hba. Скорее всего, на одном сервере для локальных подключений у вас настроено что-то вроде md5, а на другом peer

**Reply by lecavalie** [reply]
сравнил их, идентичные

**Reply by zubkov-andrei** [reply]
А какой метод авторизации там указан для подлючений local?

**Reply by lecavalie** [reply]
local   all             postgres                                peer

**Reply by zubkov-andrei** [reply, marked as answer]
Ну смотрите, у вас указан пароль в строке подключения, а метод авторизации peer не предусматривает проверки пароля - он доверяет пользователю ОС. У вас наверняка настроена авторизация по паролю для сетевых подключений - попробуйте указать параметр host=localhost в строке подключения сервера local, тогда подключение будет осуществляться через сетевой сокет и должна будет сработать авторизация по паролю.

**Reply by lecavalie** [reply]
> Ну смотрите, у вас указан пароль в строке подключения, а метод авторизации peer не предусматривает проверки пароля - он доверяет пользователю ОС. У вас наверняка настроена авторизация по паролю для сетевых подключений - попробуйте указать параметр host=localhost в строке подключения сервера local, тогда подключение будет осуществляться через сетевой сокет и должна будет сработать авторизация по паролю.

я сначала не понял, а потом как понял) благодарю
подключение состоялось
